README.md

```diff
@@ -8,14 +8,157 @@ Based on steverhoades/oauth2-openid-connect-client.
+
+```php
+use LaravelOIDCAuth\Facades\OIDC;
+// ...
+Auth::user()->refreshAccessToken();
+```
+
+Typically, you don't need to refresh the access token manually, the following doc shows an automatic way to do this.
+
+#### Auto refresh
+
+Sometimes, you want the `oidc-auth-session` to refresh the access token automatically every request, please edit
+`config/oidc_auth.php`:
+
+```php
+    // ...
+    'auto_refresh' => true,
+    // ...
+```
+
+Next time the user logged in, the `CallbackController` will set `rememberMe` option from
+`config('oidc_auth.auto_refresh')` to the guard's `login` function.
+
+### User Model & User Factory
+
+If you need an authenticated user to be an eloquent model with OIDC features, your user model must implement
+`LaravelOIDCAuth\Contracts\OIDCAuthenticatable`:
+
+```php
+namespace App\Models;
+
+use Illuminate\Notifications\Notifiable;
+use Illuminate\Contracts\Auth\MustVerifyEmail;
+use LaravelOIDCAuth\Contracts\OIDCAuthenticatable;
+use LaravelOIDCAuth\HasAccessToken;
+use LaravelOIDCAuth\RefreshesAccessToken;
+use Illuminate\Foundation\Auth\User as Authenticatable;use OpenIDConnectClient\AccessToken;
+
+class User extends Authenticatable implements OIDCAuthenticatable
+{
+    use Notifiable, HasAccessToken, RefreshesAccessToken;
+
+    // ...
+}
+```
+
+Two trait `HasAccessToken`, `RefreshesAccessToken` are added.
+
+Next, you can implement a user factory `LaravelOIDCAuth\Contracts\OIDCAuthenticatableFactory` to get user from database:
+
+```php
+namespace App;
+
+use LaravelOIDCAuth\Contracts\OIDCAuthenticatable;
+use LaravelOIDCAuth\Contracts\OIDCAuthenticatableFactory;
+use OpenIDConnectClient\AccessToken;
+
+class UserFactory implements OIDCAuthenticatableFactory
+{
+    public function authenticatable(AccessToken $token): OIDCAuthenticatable
+    {
+        $user = User::firstOrCreate(/* ... */);
+        $user->setAccessToken($token);
+        return $user;
+    }
+}
+```
+
+### Customization
+
+#### Handle OIDC callback
+
+When the OIDC Provider authenticated the user, the browser will be redirected to the callback route on your own web
+application. If you want to handle the callback request manually, you can disable the registration of default callback
+route in `config/oidc_auth.php`:
+
+```php
+    'register_default_callback_route' => false,  // default value is true, set to false
+```
+
+Next, you can create your own route & controller to handle OIDC callback, and the route must be named as
+`oidc-auth.callback`.
+
+Here is an example of your callback controller:
+
+```php
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use LaravelOIDCAuth\CallbackController;
+
+class OidcController extends CallbackController
+{
+    public function callback(Request $request)
+    {
+        // Do something
+        return parent::callback($request);
+    }
+    // You can overwrite other method from parent class LaravelOIDCAuth\CallbackController
+}
+```
+
+Add a route in `route/web.php`:
```

Comment:
Also trivial and unnecessarily assumes consumer's route organization.

Reply:
Okay.